src/components/code/code_block.tsx

```diff
@@ -351,30 +358,33 @@ const useCopy = ({
 
   const showCopyButton = isCopyable && textToCopy;
 
-  const CopyButton = () => {
-    if (!showCopyButton) return null;
-
-    return (
-      <div className="euiCodeBlock__copyButton">
-        <EuiI18n token="euiCodeBlock.copyButton" default="Copy">
-          {(copyButton: string) => (
-            <EuiCopy textToCopy={textToCopy}>
-              {(copy) => (
-                <EuiButtonIcon
-                  onClick={copy}
-                  iconType="copyClipboard"
-                  color="text"
-                  aria-label={copyButton}
-                />
-              )}
-            </EuiCopy>
-          )}
-        </EuiI18n>
-      </div>
-    );
-  };
-
-  return { innerTextRef, showCopyButton, CopyButton };
+  const _CopyButton = useMemo(() => {
+    const CopyButton: FunctionComponent = () => {
```

Comment:
shouldn't this be a `useCallback` instead of a `useMemo`?

```suggestion
  const CopyButton = useCallback(() => {
```